csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -2,29 +2,140 @@
 
 #nullable enable
 
+using Ice.Instrumentation;
+using Ice.Internal;
+using System.Diagnostics;
+using System.Net.Security;
+using System.Text;
+
 namespace Ice;
 
-public interface ObjectAdapter
+public sealed class ObjectAdapter
```

Comment:
Essentially moved all the code from ObjectAdapterI in here + cleanup + nullable enable.